### ISSUE TITLE
No results for yarn (berry) monorepo

### ISSUE THREAD

**Issue by LaurensRietveld** [IBM/audit-ci#216]
The `audit-ci` check in a Yarn (berry) monorepo may be incomplete. To reproduce:

- Use a monorepo
- Run audit-ci to only check for production dependencies
- Notice you'll see very few audit results

What happens (see [here](https://github.com/IBM/audit-ci/blob/dc9f7311b102ed7afd8db0e0927fa6f54bbcda44/lib/yarn-auditer.js#L188)) for the line involved): 
- When auditing only production dependencies, then the `--environment production` arguments  are used
- When auditing any dependency (either production or dev), then the `--all` flag is used. The `-all` flag is used in yarn to specify that we want to audit all dependencies in this workspace, and not just the one in the current working dir (see [here](https://yarnpkg.com/cli/npm/audit) for the docs)
- In other words, `audit-ci` is either 
  - checking all (both production/development) dependencies in the complete workspace
  - checking only the production dependencies of the currently focused package

It seems to me that there is a misunderstanding of the `-all` argument. I'd argue we'd always want `audit-ci` to audit the complete workspace, i.e. we should always include the `-all` argument

**Comment by quinnturner** [comment]
Thanks for the issue report. I will try to get to this on the weekend. Our current test suite does not cover monorepos. I think it should!

**Comment by quinnturner** [comment]
Hey, I haven't forgotten about this. I've had to spend all my extra time on another professional project which will be done within the week. I will try to tackle this afterwards. Thanks for understanding!

**Comment by kyletsang** [comment]
Hey @quinnturner, I got a bit of extra time and could look at this if you'd like

**Comment by quinnturner** [comment]
Hey @kyletsang, that would be great! I am still working on a mandatory side project personally this week but can do PR reviews. On my GitHub's audit-ci, I created a branch for yarn workspaces where I started creating the tests. You're welcome to look at it if you'd like or start from scratch; I only spent like 20 mins or something like that.

**Comment by quinnturner** [comment]
Released in `v6.2.0`
